server/sonar-server/src/main/java/org/sonar/server/platform/ws/ChangeLogLevelAction.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * SonarQube, open source software quality management tool.
+ * Copyright (C) 2008-2014 SonarSource
+ * mailto:contact AT sonarsource DOT com
+ *
+ * SonarQube is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * SonarQube is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.server.platform.ws;
+
+import ch.qos.logback.classic.Level;
+import org.sonar.api.server.ws.Request;
+import org.sonar.api.server.ws.Response;
+import org.sonar.api.server.ws.WebService;
+import org.sonar.api.web.UserRole;
+import org.sonar.db.Database;
+import org.sonar.server.platform.ServerLogging;
+import org.sonar.server.user.UserSession;
+
+public class ChangeLogLevelAction implements SystemWsAction {
+
+  private static final String PARAM_LEVEL = "level";
+
+  private final UserSession userSession;
+  private final ServerLogging logging;
+  private final Database db;
+
+  public ChangeLogLevelAction(UserSession userSession, ServerLogging logging, Database db) {
+    this.userSession = userSession;
+    this.logging = logging;
+    this.db = db;
+  }
+
+  @Override
+  public void define(WebService.NewController controller) {
+    WebService.NewAction newAction = controller.createAction("change_log_level")
+      .setDescription("Changes temporarily level of logs. New level is not persistent and is lost " +
+        "when restarting server.")
```

Comment:
I don't think we should give too many details on implementation. The level is applied to the loggers that we think make sense.